src/activities/Elsa.Activities.Reflection/Activities/ExecuteFunction.cs

```diff
@@ -0,0 +1,114 @@
+        public string InputVariableNames
+        {
+            get => GetState<string>(null, "InputVariableNames");
+            set => SetState(value, "InputVariableNames");
+        }
+
+        [ActivityProperty(Hint = "The name of the variable to store the returned value into.")]
+        public string OutputVariableName
+        {
+            get => GetState<string>(null, "OutputVariableName");
+            set => SetState(value, "OutputVariableName");
+        }
+
+        [ActivityProperty(Hint = "Assembly name (fullname or filename) to load")]
+        public string AssemblyName
+        {
+            get => GetState<string>(null, "AssemblyName");
+            set => SetState(value, "AssemblyName");
+        }
+
+        [ActivityProperty(Hint = "Class name to start or lookup")]
+        public string ClassName
+        {
+            get => GetState<string>(null, "ClassName");
+            set => SetState(value, "ClassName");
+        }
+
+        [ActivityProperty(Hint = "Class is a static class?")]
+        public bool IsStaticClass
+        {
+            get => GetState<bool>(null, "IsStaticClass");
+            set => SetState(value, "IsStaticClass");
+        }
+
+        [ActivityProperty(Hint = "Method name to execute")]
+        public string MethodName
+        {
+            get => GetState<string>(null, "MethodName");
+            set => SetState(value, "MethodName");
+        }
+
+        protected override async Task<ActivityExecutionResult> OnExecuteAsync(
+            WorkflowExecutionContext context,
+            CancellationToken cancellationToken)
+        {
+            await Task.Delay(1);
+            var inputValues = InputVariableNames.Split(',').Select(s => context.GetVariable(s)).ToArray();
+            return Execute(context, inputValues);
+        }
+
+        protected override ActivityExecutionResult OnResume(WorkflowExecutionContext context)
+        {
+            var inputValues = InputVariableNames.Split(',').Select(s => context.GetVariable(s)).ToArray();
+            return Execute(context, inputValues);
+        }
+
+        private ActivityExecutionResult Execute(WorkflowExecutionContext workflowContext, object[] receivedInputValues)
+        {
+            string path = Directory.EnumerateFiles(AppDomain.CurrentDomain.BaseDirectory, AssemblyName, SearchOption.AllDirectories).FirstOrDefault();
+            Assembly assembly = Assembly.LoadFrom(path);
+            object receivedOutput = null;
+
+            if (IsStaticClass)
+            {
+                var staticClassType = assembly.DefinedTypes.Where(t => t.Name == ClassName).FirstOrDefault();
+                var staticMethod = staticClassType.DeclaredMethods.Where(m => m.Name == MethodName).FirstOrDefault();
+                receivedOutput = staticMethod.Invoke(null, receivedInputValues);
+            }
+            else
+            {
+            }
```

Comment:
Looks like you can remove this block.

Reply:
No way... I have to fill it. There goes the non static handling